### PR TITLE
feat: Workflow for uploading example zip assets to releases

### LINKED ISSUE #2347: Create zip for every example on release and deploy
Automate the process of creating a zip for every example in the `examples` folder when a new release is made.
Attach each zip to the release page. Their readme needs to be added into the hugo site with a link to the release page artifact.

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

This PR adds a new github workflow that creates zip files from every directory in `/examples` and uploads them as assets to the latest or a specified release.

### Related Issues

* Closes #2347
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: .github/workflows/create-example-asset-zips.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
+name: Zip and Upload Examples
```
Comment (jeromy-cannon) [inline .github/workflows/create-example-asset-zips.yml:1]: one of the things that I've learned working on these workflows and with Nathan is that it is much better if we can run it locally for verification and testing.  Looking at your logic in this workflow, nearly all of it can move into a Taskfile, and then the workflow can call the task.  That way I, or someone else with the proper permissions can run the Taskfile for both testing, but also as a workaround in case we have pipeline issues.

Also, we want this to be part of the deploy release artifact pipeline, you have this set up as a separate workflow that needs to be called manually.

If it is only a Taskfile task, and not much else, you can just add the single job into `.github/workflows/flow-deploy-release-artifact.yaml`.  If however, it has a lot of steps, then it needs to be its own reusable workflow.

For some example logic on how to do it in a Taskfile, see: [`docs/site/Taskfile.github.yaml`](https://github.com/hiero-ledger/solo/blob/42f0275bc602e6c26e48a55b688c5f94938a3649/docs/site/Taskfile.github.yaml#L192-L192)

File: .github/workflows/flow-deploy-release-artifact.yaml [hunk 2]
```diff
@@ -201,6 +201,12 @@ jobs:
         if: ${{ github.event.inputs.dry-run-enabled != 'true' && !cancelled() && !failure() }}
         run: jf npm publish
 
+      - name: Create Example Asset Archives and Upload to Release
```
Comment (jeromy-cannon) [inline .github/workflows/flow-deploy-release-artifact.yaml:204]:
```suggestion
      - name: Create Example Asset Archives and Upload to Release
        if: ${{ github.event.inputs.dry-run-enabled != 'true' && !cancelled() && !failure() }}
```
